streams/src/test/java/org/apache/kafka/streams/processor/internals/KeyValueStoreMaterializerTest.java

```diff
@@ -117,9 +117,12 @@ public void shouldCreateKeyValueStoreWithTheProvidedInnerStore() {
         final KeyValueStoreMaterializer<String, Integer> materializer = new KeyValueStoreMaterializer<>(materialized);
         final StoreBuilder<KeyValueStore<String, Integer>> builder = materializer.materialize();
         final KeyValueStore<String, Integer> built = builder.build();
-        final StateStore inner = ((WrappedStateStore) built).inner();
+        StateStore innerMost = built;
+        while (innerMost instanceof WrappedStateStore) {
+            innerMost = ((WrappedStateStore) built).wrappedStore();
```

Comment:
`innerMost = ((WrappedStateStore) innerMost).wrappedStore();`

Even if I think, that a `WrappedStateStore` should be able to return the `wrapped` and `root` (ie, recursively wrapped store directly -- ie, there should be methods `wrapped()` and `root()`)

Reply:
What value would this provide?

Reply:
You mean type? As mentioned above, if we pass in two types to a `WrappedStateStore` we can make it type safe.